src/modules/reporting/actions/submit-market-contribute-test.js

```diff
@@ -0,0 +1,122 @@
+  const state = Object.assign({}, testState)
+  const mockSelectMarket = {}
+  mockSelectMarket.selectMarket = sinon.stub().returns(state.marketsData.testMarketId)
+  const mockStore = configureMockStore([thunk])
+  const store = mockStore(state)
+
+  const callback = sinon.stub()
+  const history = {
+    push: sinon.stub(),
+  }
+
+  afterEach(() => {
+    history.push.reset()
+    callback.reset()
+  })
+
+  after(() => {
+    submitMarketContributeReqireAPI.__ResetDependency__('getPayoutNumerators')
+  })
+
+  const augurSuccess = {
+    api: {
+      Market: {
+        contribute: (options) => {
+          options.onSent()
+          options.onSuccess()
+        },
+      },
+    },
+  }
+
+  const augurFailed = {
+    api: {
+      Market: {
+        contribute: (options) => {
+          options.onSent()
+          options.onFailed()
+        },
+      },
+    },
+  }
+
+  const getPayoutNumerators = sinon.stub().returns([10000, 0])
+  submitMarketContributeReqireAPI.__Rewire__('getPayoutNumerators', getPayoutNumerators)
+  const test = (t) => {
+    it(t.description, () => {
+      t.assertions()
+    })
+  }
+
+  describe('history is called', () => {
+    test({
+      description: `should call the expected method`,
+      assertions: () => {
+        submitMarketContributeReqireAPI.__Rewire__('augur', augurSuccess)
+        store.dispatch(submitMarketContribute('testMarketId', 0, false, 1000, history, callback))
+        assert(callback.calledOnce, `Didn't call 'callback' once as expected`)
+        assert(history.push.calledOnce, `Didn't call 'history' once as expected`)
+      },
+    })
+  })
+
+  describe('null marketId', () => {
+    test({
+      description: `should call the expected method`,
+      assertions: () => {
+        submitMarketContributeReqireAPI.__Rewire__('augur', augurSuccess)
+        store.dispatch(submitMarketContribute(null, 0, false, 1000, history, callback))
+        assert(callback.calledOnce, `Didn't call 'callback' once as expected`)
+        assert(history.push.notCalled, `Did call 'history' not expected`)
+      },
+    })
+  })
+
+  describe('empty marketId', () => {
+    test({
+      description: `should call the expected method`,
+      assertions: () => {
+        submitMarketContributeReqireAPI.__Rewire__('augur', augurSuccess)
+        store.dispatch(submitMarketContribute('', 0, false, 1000, history, callback))
+        assert(callback.calledOnce, `Didn't call 'callback' once as expected`)
+        assert(history.push.notCalled, `Did call 'history' not expected`)
+      },
+    })
+  })
+
+  describe('non number outcome', () => {
+    test({
+      description: `should call the expected method`,
+      assertions: () => {
+        submitMarketContributeReqireAPI.__Rewire__('augur', augurSuccess)
+        store.dispatch(submitMarketContribute('testMarketId', 'blah', false, 1000, history, callback))
+        assert(callback.calledOnce, `Didn't call 'callback' once as expected`)
```

Comment:
This could be broken apart to be more readable. Like: 
```
  describe('non number outcome', () => {
    beforeEach(() => {
      submitMarketContributeReqireAPI.__Rewire__('augur', augurSuccess)
      store.dispatch(submitMarketContribute('testMarketId', 'blah', false, 1000, history, callback))
    })

    it('should invoke callback', () => {
      assert(callback.calledOnce, `Didn't call 'callback' once as expected`)
    })

    it(`should not invoke history 'push' method `, () => {
      assert(history.push.notCalled, `Did call 'history' not expected`)
    })
  })
```